stdlib/asm/mem.masm

```diff
@@ -2,6 +2,118 @@ use.std::crypto::hashes::rpo
 
 # ===== MEMORY FUNCTIONS ==========================================================================
 
+#! Copies `n` felts from `read_ptr` to `write_ptr`.
+#!
+#! If possible, this procedure will copy word-aligned words internally using `memcopy_words` 
+#! procedure, decreasing the total number of cycles required to make a copy.
+#!
+#! It is advised to copy small chunks of memory using `memcopy_felts` procedure instead, since it
+#! won't have a computation logic overhead.
+#!
+#! Inputs:  [n, read_ptr, write_ptr]
+#! Outputs: []
+#!
+#! Where:
+#! - n is the number of felts which should be copied.
+#! - read_ptr is the memory pointer where the felts to copy are stored.
+#! - write_ptr is the memory pointer where the felts will be copied.
+#!
+#! Panics if:
+#! - read_ptr and write_ptr are not a u32 values.
+#!
+#! Total cycles:
+#! - if read and write pointers are mutually unaligned: 23 + 14 * num_felts
+#! - if read and write pointers are mutually aligned, it's equal to the sum of:
```

Comment:
should the first one be`aligned` and the second `unaligned`?

Reply:
I think I get the intention now, should probably be `mutually word-aligned` as defined in a further-down comment